addon/index.d.ts

```diff
@@ -20,16 +34,22 @@ export type EncapsulatedTaskDescriptorArgs<T extends EncapsulatedTaskDescriptor<
 export type EncapsulatedTaskDescriptorReturnType<T extends EncapsulatedTaskDescriptor<any, any[]>> =
   T extends { perform(...args: any[]): TaskGenerator<infer R> } ? R : unknown;
 
-/**
- * The `Task` object lives on a host Ember object (e.g.
- * a Component, Route, or Controller). You call the
- * {@linkcode Task#perform .perform()} method on this object
- * to create run individual {@linkcode TaskInstance}s,
- * and at any point, you can call the {@linkcode Task#cancelAll .cancelAll()}
- * method on this object to cancel all running or enqueued
- * {@linkcode TaskInstance}s.
- */
-export interface Task<T, Args extends any[]> {
+export type EncapsulatedTaskState<T extends object> = ExcludeKeys<
+  ExcludeKeys<T, TaskInstance<any>>,
+  { perform: any }
+>;
```

Comment:
I think instead of building custom `ExcludeKeys` and `NonOverlappingKeys` types, this could be written as:

```ts
export type EncapsulatedTaskState<T extends object> = Omit<
  T,
  'perform' | keyof TaskInstance<any>
>;
```